### ISSUE TITLE
Layer shape mark as depreecated, but heavily used in tests

### ISSUE THREAD

**Issue by Czaki** [napari/napari#1928]
## 🐛 Bug
The layer shape attribute is marked as deprecated and will be removed in 0.4.2, but it is heavily used by current tests. 

https://github.com/napari/napari/blob/5ecf0c79cd9e98fa0fa4a6030553609a220f5685/napari/layers/base/base.py#L616-L635

```
napari/layers/image/_tests/test_volume.py::test_4D_volume
  /home/czaki/Projekty/napari/napari/layers/image/_tests/test_volume.py:105: FutureWarning: The shape attribute is deprecated and will be removed in version 0.4.2. Instead you should use the extent.data and extent.world attributes to get the extent of the data in data or world coordinates.
    assert layer.shape == shape

napari/layers/image/_tests/test_volume.py::test_changing_volume
  /home/czaki/Projekty/napari/napari/layers/image/_tests/test_volume.py:121: FutureWarning: The shape attribute is deprecated and will be removed in version 0.4.2. Instead you should use the extent.data and extent.world attributes to get the extent of the data in data or world coordinates.
    assert layer.shape == shape_b
```

## To Reproduce

1. Run tests

## Expected behavior

Tests should not use deprecated parameters.

**Comment by sofroniewn** [comment]
> Tests should not use deprecated parameters.

Yes absolutely, that was flagged here #1805, and still needs to be fixed!!

**Comment by sofroniewn** [comment]
Closed by #1938
